network/modules/UDP/socket_wrapper.py

```diff
@@ -0,0 +1,102 @@
+import socket
+
+
+class UdpSocket:
+    """
+    Wrapper for Python's socket module.
+    """
+
+    def __init__(self, socket_instance: socket.socket) -> None:
+        """
+        Parameters
+        ----------
+        instance: socket.socket
+            For initializing Socket with an existing socket object.
+        """
+        if socket_instance is None:
+            self.__socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        else:
+            self.__socket = socket_instance
+
+    def send_to(self, data: bytes, address: tuple) -> bool:
+        """
+        Sends data to specified address
+
+
+        Parameters
+        ----------
+        data: bytes
+        address: tuple
+
+
+        Returns
+        -------
+        bool: if data was transferred successfully
+
+        """
+        try:
+            self.__socket.sendto(data, address)
+        except socket.error as e:
+            print(f"Could not send data: {e}")
+            return False
+
+        return True
+
+    def recv_from(self, buf_size: int) -> "tuple[bool, bytes | None]":
+        """
+        Receives data from the specified socket
+
+
+        Parameters
+        ---------
+        buf_size: int
+            The number of bytes to receive
+
+
+        Returns
+        -------
+        tuple: If the data and address were received
+
+        """
+        try:
+            data, addr = self.__socket.recvfrom(buf_size)
```

Comment:
Can you put this in a while loop? I believe that networks usually split large data into smaller packets, and may not send the whole thing in one go. Thus, you need to keep on looping until you get all of the data. Additionally, you should check that it all comes from the same address (for now, use the first recv's returned address).